src/main/java/com/keniu/validations/FieldMatchValidator.java

```diff
@@ -0,0 +1,51 @@
+package com.keniu.validations;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+/**
+ * Validator implementation for the {@code @FieldMatch} annotation.
+ */
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+
+    private static final int FIRST_CHAR_INDEX = 0;
+    private static final int MIN_LENGTH = 1;
+
+    private String firstFieldName;
+    private String secondFieldName;
+    private String message;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+        this.message = constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        try {
+            String firstValue = (String) value.getClass().getMethod("get"
+                    + capitalize(firstFieldName)).invoke(value);
+            String secondValue = (String) value.getClass().getMethod("get"
+                    + capitalize(secondFieldName)).invoke(value);
```

Comment:
```
Object fieldValue = new BeanWrapperImpl(value)
                .getPropertyValue(firstFieldName);
Object fieldMatchValue = new BeanWrapperImpl(value)
                .getPropertyValue(firstFieldName);
```
Can you simplify your solution in such a way?

Reply:
Good idea! Done